src/Riok.Mapperly/Helpers/SymbolExtensions.cs

```diff
@@ -78,6 +78,32 @@ internal static IEnumerable<IPropertySymbol> GetAllAccessibleProperties(this ITy
         return genericIntf != null;
     }
 
+
+    // TODO: refactor and rename.
+    internal static bool HasImplicitInterfaceMethod(this ITypeSymbol symbol, INamedTypeSymbol inter, string methodName)
+    {
+        // return true if symbol is the same interface - does not check that the method is implemented so class A : IList<T> { } will be accepted
+        if (SymbolEqualityComparer.Default.Equals(symbol, inter))
+        {
+            return true;
+        }
+
+        // return false if it does not implement the interface
+        if (!symbol.ImplementsGeneric(inter, out var typedInter))
+            return false;
+
+        var interfaceMethodSymbol = typedInter.GetMembers(methodName).OfType<IMethodSymbol>().Single();
+
+        var methodInterImplementaton = symbol.FindImplementationForInterfaceMember(interfaceMethodSymbol) as IMethodSymbol;
+
+        // if null then the method is unimplemented
+        if (methodInterImplementaton is null)
+            return true;
```

Comment:
Should this method really return true in case the method is not implemented at all? `HasImplicitInterfaceMethod` resolving to true for unimplemented methods seems unexpected to me. But this can never happen anyway (due to the `ImplementsGeneric` check on L92, can it? So probably just an exception could be thrown?

Reply:
Spot on, ordinarily you'd expect this method to be implemented as part of the underlying interface. But in cases like:
```C#
public class RepeatedField : ICollection<int> { }
```
The class `RepeatedField` "implements" the interface but hasn't implemented any methods. This normally fails to compile but except for the unit tests where the above example is common.

My solution is hacky but its better than updating all the tests. I'll add a comment explaining why I've done this.